netsim/addressing.py

```diff
@@ -280,33 +280,38 @@ def setup(topo: Box, defaults: Box) -> None:
 
   common.exit_on_error()
 
-def parse_prefix(prefix: typing.Union[str,dict]) -> typing.Dict:
+def parse_prefix(prefix: typing.Union[str,dict]) -> Box:
   if common.debug_active('addr'):                     # pragma: no cover (debugging printout)
     print(f"parse prefix: {prefix} type={type(prefix)}")
+
+  empty_box = Box({})
   if not prefix:
-    return {}
+    return empty_box
   supported_af = ['ip','ipv4','ipv6']
-  prefix_list: typing.Dict = {}
-  if isinstance(prefix,dict):
-    for af,pfx in prefix.items():
-      if not af in supported_af:
-        common.error( \
-          'Unknown address family %s in prefix %s' % (af,prefix), \
-          category=common.IncorrectValue,module='addressing')
-      else:
-        if isinstance(pfx,bool):
-          prefix_list[af] = pfx
-        else:
-          try:
-            prefix_list[af] = netaddr.IPNetwork(pfx)
-          except Exception as ex:
-            common.error(f'Cannot parse {af} prefix: {prefix}\n... {ex}',common.IncorrectValue,'addressing')
-            return {}
-          if str(prefix_list[af]) != str(prefix_list[af].cidr):
-            common.error(f'{af} prefix contains host bits: {prefix}',common.IncorrectValue,'addressing')
-    return prefix_list
-  else:
-    return { 'ipv4' : netaddr.IPNetwork(prefix) }
+  prefix_list = Box({})
+  if not isinstance(prefix,dict):
+    return Box({ 'ipv4' : netaddr.IPNetwork(prefix) })
+
+  for af,pfx in prefix.items():
+    if not af in supported_af:
+      common.error( \
+        'Unknown address family %s in prefix %s' % (af,prefix), \
+        category=common.IncorrectValue,module='addressing')
+      continue
+
+    if isinstance(pfx,bool):
+      prefix_list[af] = pfx
+      continue
+
+    try:
+      prefix_list[af] = netaddr.IPNetwork(pfx)
+    except Exception as ex:
+      common.error(f'Cannot parse {af} prefix: {prefix}\n... {ex}',common.IncorrectValue,'addressing')
+      return empty_box
+    if str(prefix_list[af]) != str(prefix_list[af].cidr):
```

Comment:
Without normalization this check will erroneously detect host bits.
For example, "10.00.0.0/24" != "10.0.0.0/24" so the error gets triggered

Reply:
prefix_list is already `netaddr.IPNetwork` and so `str(prefix_list)` is normalized:

```
>>> import netaddr
>>> x = netaddr.IPNetwork('10.000.00.0/24')
>>> str(x)
'10.0.0.0/24'
```

Reply:
ok, just something I hit recently